packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,375 @@
+    - [1. Composable API](#1-composable-api)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [2. Compact API](#2-compact-api)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+    - [Decision](#decision)
+      - [Only Composable API](#only-composable-api)
+      - [Only Compact API](#only-compact-api)
+      - [Both Composable \& Compact API](#both-composable--compact-api)
+      - [Conclusion](#conclusion)
+  - [Final API](#final-api)
+- [Table Library Evaluation](#table-library-evaluation)
+  - [Why a library?](#why-a-library)
+  - [What are we looking for in a library?](#what-are-we-looking-for-in-a-library)
+  - [Libraries evaluated](#libraries-evaluated)
+    - [Heavy sized library - AG Grid](#heavy-sized-library---ag-grid)
+    - [Medium sized library - Tanstack's React Table](#medium-sized-library---tanstacks-react-table)
+    - [Small sized library - React Table Library](#small-sized-library---react-table-library)
+
+
+## Features
+- Column Sorting
+- Row Selection - Single & Multiple
+- Pagination
+- Bulk Actions Toolbar
+- Horizontally Scrollable
+- Sticky Columns
+- Sticky Header
+- Sticky Footer
+- Cell Density - Normal & Comfortable
+
+## Out of scope
+We don't have enough use-cases for the following features at Razorpay and hence scoped them out of our Table component. If you are aware of any of these features being used at Razorpay, please let us know by creating an issue - we will evaluate it and scope it for future releases.
+
+- Column Reordering
+- Column Resizing
+- Column Filtering
+- Search
+- Row Expansion
+- Nested Tables
+- Hiding Columns
+- Editable Rows
+
+## Anatomy
+<img src="./table-anatomy.png" width="760" alt="Table thumbnail" />
+
+## API
+
+### API Design Approaches
+While evaluating what kind of API we want for our table, we discovered 2 approaches:
+
+#### 1. Composable API
+Composable API is a pattern where we have a set of components that can be composed together to create a table. This approach allows us to stitch together a table very similar to how you would do it with native HTML. However, it is also more verbose and requires more effort to create a table.
+
+```jsx
+// Mock data & functions
+
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const onSortChange = ({ headerKey, sortType }) => {
+  console.log({headerKey, sortType});
+};
+
+
+```
+
+
+```jsx
+// Composable Table
+
+<Table 
+    data={data} // This contains the actual data to be rendered in the table which would be retrieved from some API
+    selectionType='single' 
+    onSelectionChange={onSelectionChange} 
+    sortFunctions={sortFunctions} 
+    onSortChange={onSortChange}
+    isLoading={false}
+    cellDensity='normal'
```

Comment:
we don't have a way of customizing density per cell right? so this shall be row density rather? also maybe we can call it something more simple like row spacing

Reply:
Its not equal spacing on all sides so calling it something like `rowSpacing='spacing.5'` is misleading. We could call it `rowDensity` instead of `cellDensity` though